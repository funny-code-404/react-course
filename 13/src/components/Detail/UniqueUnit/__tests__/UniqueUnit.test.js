import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import DetailUniqueUnit from "..";

configure({ adapter: new Adapter() });

describe('DETAIL UNIQUE UNIT', () => {
    const initialState = {
        unique_unit: false,
    }

    const props = {
        data: [
            '/path/1',
        ]
    }

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<DetailUniqueUnit {...props} />);
    });

    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('initialize CreatedIn with initial state', () => {
        expect(wrapper.state()).toEqual(initialState)
    });
    it('Component is return 1 button content', () => {
        expect(wrapper.find('button')).toHaveLength(1);
    });
    it('Component should render correct data content', () => {
        const mock_data = '1'
        expect(wrapper.find('button').props().children).toEqual(mock_data);
    });
    it('Component should render correct after props changing', () => {
        const newdata = '/path/2';
        const mock_result = '2'
        wrapper.setProps({ data: [ newdata ]})
        expect(wrapper.find('button').props().children).toEqual(mock_result);
    });

    it('should return on click true',() => {
        const mock_result = true
        wrapper.find('button').simulate('click')
        expect(wrapper.state().unique_unit).toEqual(mock_result)
    });

    it('Component should render correct after changing', () => {
        const mock_result =  true;
        const mock_text = 'close';

        wrapper.setState({ unique_unit: mock_result})
        expect(wrapper.find('button').text()).toEqual(mock_text);
    });
});

// npm run test src/components/Detail/UniqueUnit