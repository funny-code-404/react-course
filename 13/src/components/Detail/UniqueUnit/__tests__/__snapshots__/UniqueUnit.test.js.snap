// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DETAIL UNIQUE UNIT to Match Snapshot 1`] = `
<div
  key="/path/1"
>
  <p>
    Unique unit:
  </p>
  <button
    onClick={[Function]}
  >
    1
  </button>
</div>
`;

exports[`UniqueUnit to Match Snapshot 1`] = `
<div
  key="/path/1"
>
  <p>
    Unique unit:
  </p>
  <button
    onClick={[Function]}
  >
    1
  </button>
</div>
`;
